application/migrations/010_add_long_running_flag.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class Migration_add_long_running_flag extends CI_Migration
+{
+    // Add a column to indicate whether the office's crawl takes longer than 23 hours
+    public function up()
+    {
+        // add our column if it doesn't exist
+        $this->CI->db->query("CALL add_column_if_not_exists('offices', 'long_running', 'varchar(256)')");
+        // test if it exists
+        $query = $this->CI->db->query("SELECT column_exists('offices', 'long_running')");
+        $row = $query->first_row('array');
+        $tableAndColumnExist = (bool) array_shift($row);
+        // add values if it exists
+        if ($tableAndColumnExist) {
```

Comment:
You don't need this test code! (The test code for the stored proc does this, but that's just because it's testing that the stored proc has the expected behavior. You can always assume that `add_column_if_not_exists` succeeds. That's why we use it; otherwise the migration might get stuck continuously failing when reattempted after a single failure.)